### PR TITLE
Add setting to only analyze open documents for references

### PR DESCRIPTION
**Pull request by SeeminglyScience**
## PR Summary

Dependent on PowerShell/PowerShellEditorServices#1917

## PR Checklist

Note: Tick the boxes below that apply to this pull request by putting an `x` between the square brackets.
Please mark anything not applicable to this PR `NA`.

- [x] PR has a meaningful title
- [x] Summarized changes
- [N/A] PR has tests
- [x] This PR is ready to merge and is not work in progress
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready

### CODE REVIEW

File: src/settings.ts [hunk 1]
```diff
@@ -102,6 +102,7 @@ export interface ISettings {
     cwd?: string;
     notebooks?: INotebooksSettings;
     enableReferencesCodeLens?: boolean;
+    analyzeOpenDocumentsOnly?: boolean;
```
Comment (SeeminglyScience) [inline src/settings.ts:105]: btw I picked this name specifically to match C#'s

It doesn't perfectly fit, but I think it's pretty accurate. Open to changing though.

Comment (andschwa) [reply src/settings.ts:105]: LGTM.

File: src/settings.ts [hunk 2]
```diff
@@ -273,6 +274,8 @@ export function load(): ISettings {
             configuration.get<string>("cwd", undefined),
         enableReferencesCodeLens:
             configuration.get<boolean>("enableReferencesCodeLens", true),
+        analyzeOpenDocumentsOnly:
+            configuration.get<boolean>("analyzeOpenDocumentsOnly", true),
```
Comment (andschwa) [inline src/settings.ts:278]: Don't we need a bit in `package.json` too?

Comment (SeeminglyScience) [reply src/settings.ts:278]: lol sure do. My b, I'll commit that in a sec. Good catch!

**Review by andschwa — APPROVED** [review]

**Pull request merged**